topbin/coqtop_byte_bin.ml

```diff
@@ -17,18 +17,17 @@ let _ = CErrors.register_handler (function
   )
 
 let drop_setup () =
-  begin try
+  begin
     (* Enable rectypes in the toplevel if it has the directive #rectypes *)
-    begin match Hashtbl.find Toploop.directive_table "rectypes" with
-      | Toploop.Directive_none f -> f ()
-      | _ -> ()
-    end
-    with
-    | Not_found -> ()
+    match Toploop.get_directive "rectypes" with
+    | None -> ()
+    | Some (Toploop.Directive_none f) -> f ()
+    | Some _ ->
+      Format.eprintf "warning: rectypes directive has changed!!@\n%!"
```

Comment:
What's going on with the end of the sentence? Also perhaps:
```suggestion
      Format.eprintf "Warning: rectypes directive has changed!!@\n%!"
```

Reply:
Thanks!